src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -305,6 +307,22 @@ MergeTreeData::MergeTreeData(
     if (!canUsePolymorphicParts(*settings, &reason) && !reason.empty())
         LOG_WARNING(log, "{} Settings 'min_rows_for_wide_part', 'min_bytes_for_wide_part', "
             "'min_rows_for_compact_part' and 'min_bytes_for_compact_part' will be ignored.", reason);
+
+    common_assignee_trigger = [this] (bool delay) noexcept
```

Comment:
This callbacks look very similar and I don't understand how future implementation will look like. Maybe you can provide some example of the callback from future?

Reply:
This callback will be passed to task's constructor. The main reason to move this callback into task is that each task is associated with a storage. And commonly has `MergeTreeData &` field. We could pass this callback directly to an `MergeTreeBackgroundExecutor` through a `BackgroundJobsAssignee` or we can pass `MergeTreeData &` and make an executor to be a friend of `MergeTreeData`, but this will be too complex.